src/test/java/org/jabref/AutomaticRelinkTest.java

```diff
@@ -0,0 +1,174 @@
+package org.jabref;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.LinkedFile;
+import org.jabref.model.metadata.MetaData;
+import org.jabref.preferences.FilePreferences;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class AutomaticRelinkTest {
+
+    private Path main;
+    private Path B;
+    private Path A;
+    private Path beforeFilePath;
+    private BibEntry entry;
+    private FilePreferences filePreferences;
+    private BibDatabaseContext databaseContext;
+    private String defaultDirectory;
+
+    /**
+     * `MoveFilesCleanupTest` Used as an example of how to do this
+     *
+     * @param bibFolder
+     * @throws IOException
+     */
+    @BeforeEach
+    void setUp(@TempDir Path bibFolder) throws IOException {
+        // Create Main
```

Comment:
Please remove these kind of comments. The line after the next one even contains the same words!